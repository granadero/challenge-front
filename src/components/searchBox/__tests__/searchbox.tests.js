import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import { useRouter } from "next/navigation";
import SearchBox from "../searchBox";

const mockRouterReplace = jest.fn();
const mockRouterRefresh = jest.fn();
const mockRouterPush = jest.fn();

jest.mock("next/navigation", () => {
  const originalModule = jest.requireActual("next/navigation");
  return {
    __esModule: true,
    ...originalModule,
    useRouter: jest.fn().mockImplementation(() => {
      return {
        push: mockRouterPush,
        replace: mockRouterReplace,
        refresh: mockRouterRefresh,
      };
    }),
    useSearchParams: jest.fn().mockImplementation(() => {
      return new URLSearchParams(window.location.search);
    }),
    usePathname: jest.fn().mockImplementation((pathArgs) => {
      return pathArgs;
    }),
  };
});

describe("SearchBox", () => {
  afterEach(() => {
    mockRouterRefresh.mockRestore();
    mockRouterReplace.mockRestore();
    mockRouterPush.mockRestore();
  });
  it("updates inputValue when input field changes", () => {
    const { getByPlaceholderText } = render(<SearchBox />);
    const inputField = getByPlaceholderText("Buscar...");
    fireEvent.change(inputField, { target: { value: "test" } });
    expect(getByPlaceholderText("Buscar...")).toHaveValue("test");
  });

  it("calls onSearch when Enter key is pressed", () => {
    const { getByPlaceholderText } = render(<SearchBox />);
    const inputField = getByPlaceholderText("Buscar...");
    fireEvent.change(inputField, { target: { value: "test" } });
    fireEvent.keyDown(inputField, { key: "Enter", keyCode: 13 });
    expect(useRouter().push).toHaveBeenCalledTimes(1);
  });

  it("does not call onSearch when input field is empty", () => {
    const { getByPlaceholderText } = render(<SearchBox />);
    const inputField = getByPlaceholderText("Buscar...");
    fireEvent.keyDown(inputField, { key: "Enter", keyCode: 13 });
    expect(useRouter().push).not.toHaveBeenCalled();
  });

  it("calls router.push with correct URL when onSearch is called", () => {
    const { getByPlaceholderText } = render(<SearchBox />);
    const inputField = getByPlaceholderText("Buscar...");
    fireEvent.change(inputField, { target: { value: "test" } });
    fireEvent.keyDown(inputField, { key: "Enter", keyCode: 13 });
    expect(useRouter().push).toHaveBeenCalledWith("/items/?search=test");
  });

  it("should match snapshot", () => {
    const { container } = render(<SearchBox />);
    expect(container).toMatchSnapshot();
  });
});
