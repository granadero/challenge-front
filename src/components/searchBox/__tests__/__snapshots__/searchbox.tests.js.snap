// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox should match snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <img
        alt="Logo Mercado Libre"
        class="logo"
        data-nimg="1"
        decoding="async"
        height="34"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
        style="color: transparent;"
        width="134"
      />
      <div
        class="inputContainer"
      >
        <input
          placeholder="Buscar..."
          type="text"
          value=""
        />
        <button
          type="submit"
        >
          <img
            alt="BotÃ³n buscar"
            data-nimg="1"
            decoding="async"
            height="16"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
            style="color: transparent;"
            width="16"
          />
        </button>
      </div>
    </div>
  </header>
</div>
`;
